RocketChat/Rocket.Chat
# fix: rejected conference calls continue to ring

## Proposed changes (including videos or screenshots)





## Issue(s)




## Steps to test or reproduce

1. Have someone start a direct call with you.
2. Decline the call.

Expected behavior:
- The ringer would disappear and the ringing tone would stop playing

Actual behavior:
- Sometimes the ringer stops and starts again immediatelly.

Notes:
- It might be easier to trigger the problem on slower connections.
- The problem is more likely to happen if you decline the call approximatelly 2.5 or 5.5 seconds after it started ringing.

## Further comments


Back when the videoconf ringer was first implemented, we kept a list of muted calls separately from the call data itself. That was changed in a refactor, but since all call information is dropped when the user rejects it, the videoconf manager would think it was receiving a new call if it continued to receive ringing notifications from the caller.

So a race condition between the ringing notifications and our rejection notification could cause the ringing to restart for up to 10 seconds. Also if the remote user failed to receive our rejection, they would not stop sending new ringing notifications and the local client would continue ringing for up to 30 seconds.
